### DISCUSSION TITLE
Support for `device_mode`

### DISCUSSION THREAD

**Discussion by MoonshineSG** [Koenkk/zigbee2mqtt#11732] [General]
If I'm not wrong "device_mode" allows cheesing of different kind of modes for the physical switch connected to a mini module (Lonsonho TS110E_2gang in my case)

But ... that is not shown in the z2m gui. Is there a way to add that in ?

**Comment by ndbroadbent** [comment]
I am really interested to learn more about this device_mode option. Where can I find some information about this?

**Reply by MoonshineSG** [reply]
HA allows you to change the device type, so the function doesn't need to be replicated here.

**Reply by ndbroadbent** [reply]
@MoonshineSG This issue was about the "switch mode" option for Lonsonho TS110E_2gang. I also had the same question for my dimmer modules: https://github.com/Koenkk/zigbee2mqtt/discussions/12684

I discovered that the “switch_type” option actually was added recently, in this PR: https://github.com/Koenkk/zigbee2mqtt/issues/12952

So you should now see "switch_type" in z2m for your Lonsonho TS110E_2gang, with options for toggle/state/momentary:

![image](https://user-images.githubusercontent.com/139536/192128030-87ee3414-5f5f-4054-9df6-cc4890f0e7ac.png)

So I believe this issue can be closed now
